src/docs/development/platform-integration/add-to-app-android/add-flutter-fragment.md

```diff
@@ -0,0 +1,311 @@
+    // be sure to override this method and return "true" so that this
+    // FlutterFragment does not destroy your FlutterEngine when the Fragment
+    // is destroyed.
+    @Override
+    protected boolean retainFlutterEngineAfterFragmentDestruction() {
+        return true;
+    }
+}
+```
+
+By providing a pre-warmed `FlutterEngine` as shown above, your app renders the
+first Flutter frame as quickly as possible.
+
+## Display a splash screen
+
+The initial display of Flutter content requires some amount of time, even if a
+pre-warmed `FlutterEngine` is used. To help improve the user experience around
+this brief waiting period, Flutter supports the display of a splash screen until
+Flutter renders its first frame. For instructions about showing a splash screen,
+please see the [Android splash screen guide].
+
+[Android splash screen guide]: /docs/development/platform-integration/add-to-app-android/splash-screens
+
+## Run Flutter with a specified initial route
+
+An Android app might contain many independent Flutter experiences, running in
+different `FlutterFragment`s, with different `FlutterEngine`s. In these
+scenarios, it is common for each Flutter experience to begin with different
+initial routes (routes other than `/`). To facilitate this, `FlutterFragment`'s
+`Builder` allows you to specify a desired initial route, as shown below:
+
+```java
+FlutterFragment flutterFragment = new FlutterFragment.Builder()
+    .initialRoute("myInitialRoute/")
+    .build();
+```
+
+TODO: Likely reader question: what if I'm using the same FlutterEngine across
+FlutterFragments?
+
+## Run Flutter from a specified entrypoint
+
+Similar to varying initial routes, different `FlutterFragments` may wish to
+execute different Dart entrypoints. In a typical Flutter app there is only one
+Dart entrypoint: `main()`, but you can define other entrypoints. See [defining
+Dart entrypoints] for instructions on defining such entrypoints.
+
+[defining Dart entrypoints]: /docs/development/platform-integration/add-to-app-android/custom-dart-entrypoints
+
+`FlutterFragment` supports specification of the desired Dart entrypoint to
+execute for the given Flutter experience. To specify an entrypoint, build
+`FlutterFragment` as follows:
+
+```java
+FlutterFragment flutterFragment = new FlutterFragment.Builder()
+    .dartEntrypoint("mySpecialEntrypoint")
+    .build();
+```
+
+The above `FlutterFragment` configuration results in the execution of a Dart
+entrypoint called `mySpecialEntrypoint()`. Notice that the parentheses `()` are
+not included in the `dartEntrypoint` `String` name.
+
+## Control `FlutterFragment`'s render mode
+
+`FlutterFragment` can either use a `SurfaceView` to render its Flutter content,
+or it can use a `TextureView`. The default is `SurfaceView`, which is
+significantly better for performance than `TextureView`. However, `SurfaceView`
+cannot be used for Android platform animations, nor can it be interleaved in the
```

Comment:
I think we chatted about this before but I don't think this is true anymore past O

Reply:
Do you remember where some docs are on this to verify?

Reply:
I don't know if there are longer form articles. See first info box in https://developer.android.com/reference/android/view/SurfaceView

Reply:
Updated the description of limitations.